src/4_mockFetch.test.js

```diff
@@ -1,18 +1,33 @@
 import { fetchJoke } from "./4_mockFetch";
 
 describe("4_mockFetch_test", () => {
-  it("should fetch API successfully", async () => {
-    const result = await fetchJoke();
-    expect(result).toBe(true);
+  global.fetch = jest.fn(() => {
```

Comment:
bug: this mock is incorrect, you didn't copy it correctly from the tutorial. [screenshot](https://github.com/TingChiehLin/testing_practice/assets/14869087/32db1c52-2991-4d2b-8f04-530b367a00d4). It's missing a Promise inside json.

It should be like this:

```js
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () =>
        Promise.resolve({ /* the response here*/ }),
    })
  );
```